tests/unit/igraph_steiner_tree_fpt.c

```diff
@@ -0,0 +1,69 @@
+
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+int main()
+{
+    igraph_t g_empty, g_empty_dir, g_lm;
+
+    igraph_vector_t steiner_terminals;
+    igraph_vector_t weights_empty, weights_lm;
+    
+    igraph_vector_init(&weights_empty, 0);
+
+    igraph_vector_init(&steiner_terminals, 0);
+
+    VECTOR(steiner_terminals)
+    [1] = 2;
+    VECTOR(steiner_terminals)
+    [2] = 4;
+
+    igraph_vector_init_int(&weights_lm, 0);
```

Comment:
Did you mean to use `vector_init` instead of `vector_init_int`? Also, this vector is initialized to size zero as well, then you write into it.